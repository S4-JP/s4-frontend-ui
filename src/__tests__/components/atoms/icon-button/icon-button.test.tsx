import React from 'react'
import { render } from '@testing-library/react'
import { IconButton, Color } from '../../../../index'

/**
 * type Props = {
 *   focus?: boolean
 *   children: React.ReactNode
 *   handleClick?(e: React.MouseEvent<HTMLElement>): void
 * }
 */
const Dummy = () => <span>dummy</span>
describe('IconButtonのレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <IconButton>
        <Dummy />
      </IconButton>
    )

    expect(getByTestId('atoms-ib-wrap')).toMatchSnapshot()
  })

  test('props.focus=trueに設定した時、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <IconButton focus={true}>
        <Dummy />
      </IconButton>
    )

    expect(getByTestId('atoms-ib-wrap')).toHaveStyle(
      `background-color: ${Color.PRIMARY._500};`
    )
  })
})
