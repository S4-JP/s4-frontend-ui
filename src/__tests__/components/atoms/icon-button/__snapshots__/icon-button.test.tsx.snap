// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonのレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-fEOsli cEXtHP"
  data-testid="atoms-ib-wrap"
>
  <span>
    dummy
  </span>
</div>
`;
